plugin_ide.ui/src/com/github/rustdt/ide/ui/editor/RustEditorActionContributor.java

```diff
@@ -28,6 +51,99 @@ protected RustOpenDefinitionOperation createOpenDefinitionOperation(ITextEditor
+				try {
+					ResourcesPlugin.getWorkspace().getRoot().accept(RustEditorActionContributor.this::addResource);
+				} catch(Exception e) {
+					AbstractLangCore.log().logError("Global source structure could not be determined", e);
+				}
+				return Status.OK_STATUS;
+			}
+		}.schedule();
+	}
+	
+	private boolean addResource(IResource resource) {
+		convertToRustFileLocation(resource).ifPresent(RustEditorActionContributor::fileTouched);
+		return true;
+	}
+	
+	private void listenToUpdatesOfGlobalSourceStructure() {
+		ResourcesPlugin.getWorkspace().addResourceChangeListener(this::processEvent, IResourceChangeEvent.POST_CHANGE);
 	}
 	
+	private void processEvent(IResourceChangeEvent event) {
+		new Job("Updating types") {
+			@Override
+			protected IStatus run(IProgressMonitor monitor) {
+				try {
+					event.getDelta().accept(RustEditorActionContributor.this::applyResourceDelta);
+				} catch(Exception e) {
+					AbstractLangCore.log().logError("Global source structure could not be determined", e);
+				}
+				return Status.OK_STATUS;
+			}
+		}.schedule();
+	}
+	
+	private boolean applyResourceDelta(IResourceDelta delta) {
+		convertToRustFileLocation(delta.getResource()).ifPresent(location -> {
+			switch(delta.getKind()) {
+			case IResourceDelta.ADDED:
+			case IResourceDelta.CHANGED:
+				fileTouched(location);
+				break;
+			case IResourceDelta.REMOVED:
+				fileRemoved(location);
+				break;
+			}
+		});
+		return true;
+	}
+	
+	private static Optional<Location> convertToRustFileLocation(IResource resource) {
+		return Optional.of(resource)
+				.filter(res -> res instanceof IFile)
+				.filter(RustEditorActionContributor::isValidRustPath)
+				.map(ResourceUtils::getResourceLocation);
+	}
+	
+	// TODO: Find a good filter
+	private static boolean isValidRustPath(IResource resource) {
+		String path = resource.getFullPath().toString();
+		return path.contains("src") && path.endsWith(".rs");
+	}
+	
+	private static void fileRemoved(Location location) {
+		System.out.println("RustEditorActionContributor - File removed: " + location);
+		GlobalSourceStructure.fileRemoved(location);
+	}
+	
+	private static void fileTouched(Location location) {
+		System.out.println("RustEditorActionContributor - File touched: " + location);
+		try {
+			String source = FileUtil.readFileContents(location, StringUtil.UTF8);
+			RustParseDescribeLauncher parseDescribeLauncher = new RustParseDescribeLauncher(LangCore.getToolManager(), () -> false);
+			String parseDescribeStdout = parseDescribeLauncher.getDescribeOutput(source, location);
+			
+			RustParseDescribeParser parseDescribeParser = new RustParseDescribeParser(location, source);
+			SourceFileStructure fileStructure = parseDescribeParser.parse(parseDescribeStdout);
+			
+			if(fileStructure.getParserProblems().isEmpty()) {
+				GlobalSourceStructure.fileTouched(location, fileStructure);
+			}
+		} catch(OperationCancellation | CommonException e) {
+			throw new RuntimeException(e);

```

Comment:
It's a problem that has to handled somehow. One basic solution is indeed logging then, for example using `LangCore.logError`. This is will show up in the Eclipse error log. It's a bit basic because it's hard to display that as an error somewhere else in the UI. 

A better solution would be to store the error in `GlobalSourceStructure` somehow, associated with Location. For example the SourceModelManager stores for each location not just a `SourceFileStructure` but a  `CommonResult<SourceFileStructure>` which means it can store the error for that location, and so we are able to display the error in the outline view. But I think all this can be added later, for now logging would be fine.